apps/ocas/src/targets.erl

```diff
@@ -0,0 +1,278 @@
+    spawn_target( {ipv6_address, Ipv6Address}, Req, State3 );
+
+handle_address(AddressType, _Specifiers, Req, State ) ->
+    %% unknown address type
+    lager:info("~p Don't know this Address type", [AddressType] ),
+    {ok, Req2} = cowboy_req:reply( 400
+                                 , []
+                                 , <<"Don't know Address Type">>
+                                 , Req
+                                 ),
+
+    %% don't continue on, return because of unexpected response
+    {ok, Req2, State}.
+
+%% spawn target servers
+spawn_target( {hostname, HostName}, Req, State ) ->
+    State2 = maps:put(target, hostname, State),
+    State3 = maps:put(hostname, HostName, State2),
+    %% start gen_server for that target
+    {ok, Pid} = tgt_hostname:start(State),
+    State4 = maps:put(target_pid, Pid, State3),
+
+    %% check with keep alive
+    TargetKeepAlive = tgt_hostname:keepalive(),
+    lager:debug("TargetKeepAlive: ~p ", [TargetKeepAlive]),
+
+    %% tail end recurse
+    target_valid({hostname, HostName}, TargetKeepAlive, Req, State4);
+
+spawn_target( {ipv4_address, Ipv4Address}, Req, State ) ->
+    State2 = maps:put(target, ipv4_address, State),
+    State3 = maps:put(ipv4_address, Ipv4Address, State2),
+    %% start gen_server for that target
+    {ok, Pid} = tgt_ipv4_address:start(State),
+    State4 = maps:put(target_pid, Pid, State3),
+
+    %% check with keep alive
+    TargetKeepAlive = tgt_ipv4_address:keepalive(),
+    lager:debug("TargetKeepAlive: ~p ", [TargetKeepAlive]),
+
+    %% tail end recurse
+    target_valid({ipv4_address, Ipv4Address}, TargetKeepAlive, Req, State4);
+
+spawn_target( {ipv6_address, Ipv6Address}, Req, State ) ->
+    State2 = maps:put(target, ipv6_address, State),
+    State3 = maps:put(ipv6_address, Ipv6Address, State2),
+    %% start gen_server for that target
+    {ok, Pid} = tgt_ipv6_address:start(State),
+    State4 = maps:put(target_pid, Pid, State3),
+
+    %% check with keep alive
+    TargetKeepAlive = tgt_ipv6_address:keepalive(),
+    lager:debug("TargetKeepAlive: ~p ", [TargetKeepAlive]),
+
+    %% tail end recurse
+    target_valid({ipv6_address, Ipv6Address}, TargetKeepAlive, Req, State4);
+
+spawn_target( {network_firewall, NetworkFirewall}, Req, State ) ->
+    State2 = maps:put(target, network_firewall, State),
+    State3 = maps:put(network_firewall, NetworkFirewall, State2),
+    %% start gen_server for that target
+    {ok, Pid} = tgt_network_firewall:start(State),
+    State4 = maps:put(target_pid, Pid, State3),
+
+    %% check with keep alive
+    TargetKeepAlive = tgt_network_firewall:keepalive(),
+    lager:debug("TargetKeepAlive: ~p ", [TargetKeepAlive]),
+
+    %% tail end recurse
+    target_valid({network_firewall, NetworkFirewall}, TargetKeepAlive, Req, State4);
+
+spawn_target( {TargetType, Value},  Req, State ) ->
+    lager:info("spawn_target - no function head for ~p; Value=~p", [TargetType, Value]),
```

Comment:
Line 237 is too long:     lager:info("spawn_target - no function head for ~p; Value=~p", [TargetType, Value]),.